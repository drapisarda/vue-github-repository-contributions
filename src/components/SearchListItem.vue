<template>
  <li
    :class="customClasses"
    @click="$emit('click', name)"
    class="list-group-item list-group-item-action"
  >
    {{ name }}
  </li>
</template>

<script>
export default {
  name: 'SearchListItem',
  props: {
    name: {
      type: String,
      required: true,
    },
    active: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
  computed: {
    customClasses() {
      return {
        active: this.active,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
li {
  &:hover {
    cursor: pointer;
  }
}
</style>
