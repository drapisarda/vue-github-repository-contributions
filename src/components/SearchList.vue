<template>
  <ul class="list-group">
    <search-list-item
      v-for="repository in repositories"
      :key="repository"
      :name="repository"
      @click="handleItemClick"
    />
  </ul>
</template>

<script>
import SearchListItem from '@/components/SearchListItem.vue';

export default {
  name: 'SearchList',
  components: {
    SearchListItem,
  },
  props: {
    repositories: {
      type: Array,
      requierd: false,
      default: () => [],
    },
  },
  methods: {
    handleItemClick(repository) {
      this.$store.dispatch('SEARCH_CONTRIBUTORS', repository);
    },
  },
};
</script>
