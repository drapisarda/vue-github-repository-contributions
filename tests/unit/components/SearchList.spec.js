import { shallowMount } from '@vue/test-utils';
import SearchList from '@/components/SearchList.vue';
import SearchListItem from '@/components/SearchListItem.vue';

describe('SearchList', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SearchList);
  });

  it('renders component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('does not render children when "repositories" is empty', () => {
    wrapper.setProps({
      repositories: [],
    });

    const searchListItems = wrapper.findAll(SearchListItem);

    expect(searchListItems.length).toBe(0);
  });

  it('render children when "repositories" is not empty', () => {
    wrapper.setProps({
      repositories: ['repo1', 'repo2'],
    });

    const searchListItems = wrapper.findAll(SearchListItem);

    expect(searchListItems.length).toBe(2);
  });

  it('handle searchListItem emit click', () => {
    const mock = jest.fn();
    const repository = 'some-repo-name';

    wrapper.setProps({
      repositories: [repository, 'repo2'],
    });

    const searchListItem = wrapper.find(SearchListItem);

    wrapper.setMethods({
      handleItemClick: mock,
    });

    searchListItem.vm.$emit('click', repository);

    expect(mock).toHaveBeenCalledWith(repository);
  });
});
