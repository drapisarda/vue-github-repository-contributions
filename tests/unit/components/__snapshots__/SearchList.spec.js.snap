// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchList renders component 1`] = `<ul class="list-group"></ul>`;
